apps/studio/src/components/ResourceSelector/ResourceSelectorContent.tsx

```diff
@@ -0,0 +1,150 @@
+import { ResourceItem, ResourceItemSkeleton } from "./ResourceItem"
+import { lastResourceItemInAncestryStack } from "./utils"
+
+const NoItemsInFolderResult = () => {
+  return (
+    <Text
+      h="full"
+      w="full"
+      textAlign="center"
+      textStyle="caption-2"
+      display="flex"
+      alignItems="center"
+      justifyContent="center"
+    >
+      This folder is empty.
+    </Text>
+  )
+}
+
+const ResourceItemsResults = ({
+  resourceItemsWithAncestryStack,
+  isResourceIdHighlighted,
+  isResourceItemDisabled,
+  hasAdditionalLeftPadding,
+  handleClickResourceItem,
+}: {
+  resourceItemsWithAncestryStack: ResourceItemContent[][]
+  isResourceIdHighlighted: (resourceId: string) => boolean
+  isResourceItemDisabled: (resourceItem: ResourceItemContent) => boolean
+  hasAdditionalLeftPadding: boolean
+  handleClickResourceItem: (
+    resourceItemWithAncestryStack: ResourceItemContent[],
+  ) => void
+}) => {
+  return resourceItemsWithAncestryStack.map((resourceItemWithAncestryStack) => {
+    const lastChild = lastResourceItemInAncestryStack(
+      resourceItemWithAncestryStack,
+    )
+
+    if (!lastChild) {
+      throw new Error(
+        "Unexpected undefined lastChild from lastResourceItemInAncestryStack",
+      )
+    }
+
+    return (
+      <ResourceItem
+        key={lastChild.id}
+        item={lastChild}
+        isDisabled={isResourceItemDisabled(lastChild)}
+        isHighlighted={isResourceIdHighlighted(lastChild.id)}
+        handleOnClick={() =>
+          handleClickResourceItem(resourceItemWithAncestryStack)
+        }
+        hasAdditionalLeftPadding={hasAdditionalLeftPadding}
+      />
+    )
+  })
+}
+
+const ZeroResult = ({
+  searchQuery,
+  handleClickClearSearch,
+}: {
+  searchQuery: string
+  handleClickClearSearch: () => void
+}) => {
+  return (
+    <VStack
+      h="full"
+      w="full"
+      alignItems="center"
+      justifyContent="center"
+      gap="0.75rem"
+    >
+      <VStack
+        w="full"
+        alignItems="center"
+        justifyContent="center"
+        gap="0.25rem"
+      >
+        <Text textStyle="subhead-2" textAlign="center">
+          We can't find anything with
+          <br />"{searchQuery}" in title
+        </Text>
+        <Text textStyle="caption-2">Try searching for something else.</Text>
+      </VStack>
+      <Button variant="link" size="xs" onClick={handleClickClearSearch}>
+        <Text textStyle="caption-2">Clear search</Text>
+      </Button>
+    </VStack>
+  )
+}
+
+export const LoadingResourceItemsResults = () => {
+  return Array.from({ length: 5 }).map((_, index) => (
+    <ResourceItemSkeleton key={`loading-${index}`} />
+  ))
+}
+
+export const SuspendableContent = ({
+  resourceItemsWithAncestryStack,
+  isResourceIdHighlighted,
+  isResourceItemDisabled,
+  hasAdditionalLeftPadding,
+  handleClickResourceItem,
+  isSearchQueryEmpty,
+  searchQuery,
+  clearSearchValue,
+  isLoading,
```

Comment:
this component is getting very bloated already - we used to have this issue in Classic, where components had 10+ props and took on a load of responsibilities, which in turn limited dev speed as we were afraid of breaking something accidentally. 

not going to block this PR because i think the functionality should come first but it would be good if we could separate out these props (if possible) in a future PR